src/map/skill.cpp

```diff
@@ -2687,6 +2705,15 @@ int skill_counter_additional_effect (struct block_list* src, struct block_list *
 		}
 	}
 
+	// Pet Autobonus when attacked
+	if (dstsd != nullptr && dstsd->pd != nullptr && dstsd->pd->autobonus2 != nullptr && !status_isdead(bl) && !(skill_id && skill_get_nk(skill_id, NK_NODAMAGE))) {
```

Comment:
Would be better to merge the two ifs for normal autobonus and these.
Like combine one outer if with all checks that apply to both and then have inner ifs for pc bonus and pet bonus.
This comment most likely to all of these checks.